app/src/main/java/ir/cafebazaar/poolakeysample/MainActivity.kt

```diff
@@ -84,6 +90,40 @@ class MainActivity : AppCompatActivity() {
                 payment.getSubscribedProducts(handlePurchaseQueryCallback())
             }
         }
+        setGetSkuDetailClickListener()
+    }
+
+    private fun setGetSkuDetailClickListener() {
+
+        getSkuDetailInAppButton.setOnClickListener {
+            if (paymentConnection.getState() == ConnectionState.Connected) {
```

Comment:
In order to have better readability, maintainability, and have a smaller method I suggest moving this part to another method and call that method here.
@shayanpourvatan